### PR TITLE
Issue #34 enhance cmdfile functionality

### PR DESCRIPTION
**Pull request by calebcscott**
Ensure proper error checking throughout each function call
as well as proper debug output messages. Some warnings still
exist in test_command where overflow could occur. Update bot.c
to utilize new function calls and comment out old ones. Update
headfile to show new prototypes available outside cmdfile.c.

Tested production code and ensured that basic functionality
still exists and added ability to add new command without
re-read of harddisk.

Removed Debug output from bot.c and commented out DEBUG macro
to prevent DEBUG output in production code

### CODE REVIEW

**Comment by calebcscott** [comment]
Also solves Issue #35 as cmdfile.c abstracts away reading/writing to command file for bot.c

File: cmdfile.c [hunk 1]
```diff
@@ -1,29 +1,113 @@
...
+
+/***************** read_commands ************************************
+ * Purpose:
+ *  To open and read either user defined command file or default
+ *      "commands.csv" file into array struct of Commands
+ * 
+ * Params:
+ *  I char  *filename       string of user defined filename
+ * 
+ * Returns:
```
Comment (SamBkamp) [inline cmdfile.c:86]: Maybe we could return -1 on error and set errno istead? Seems more professional.

File: cmdfile.c [hunk 2]
```diff
@@ -36,46 +120,233 @@ int init() {
...
+ *  I/O int         maxOutputLen    length available for outputMsg buffer   
+ * 
+ * Returns:
+ *  0 - Command already exists/Null pointer to output/output size to small
+ *  1 - Message copied successfully to outputMsg
+ * 
+ * Notes/Warnings:
+ *  If maxOutputLen is greater than actual allocated memory size for 
+ *      outputMsg buffer strncpy will overflow buffer and could cause
+ *      undefined behavior - ensure checking of error codes and buffer sizes
```
Comment (SamBkamp) [inline cmdfile.c:280]: maybe we could printf with a warning if maxOutputLen > sizeof(outputMsg)

**Review by SamBkamp — CHANGES_REQUESTED** [review]
good change, Just one thing to iron out. On program quit there seems to be a double free as per these logs;

```
./Aladdin --join botbkamp
:tmi.twitch.tv 001 botbkamp :Welcome, GLHF!
:tmi.twitch.tv 002 botbkamp :Your host is tmi.twitch.tv
:tmi.twitch.tv 003 botbkamp :This server is rather new
:tmi.twitch.tv 004 botbkamp :-
:tmi.twitch.tv 375 botbkamp :-
:tmi.twitch.tv 372 botbkamp :You are in a maze of twisty passages, all alike.
:tmi.twitch.tv 376 botbkamp :>
:botbkamp!botbkamp@botbkamp.tmi.twitch.tv JOIN #botbkamp
:botbkamp.tmi.twitch.tv 353 botbkamp = #botbkamp :botbkamp
:botbkamp.tmi.twitch.tv 366 botbkamp #botbkamp :End of /NAMES list
:bkamp_!bkamp_@bkamp_.tmi.twitch.tv PRIVMSG #botbkamp :!project
:bkamp_!bkamp_@bkamp_.tmi.twitch.tv PRIVMSG #botbkamp :!credits
[#botbkamp]> addcmd !hey hello o/
:bkamp_!bkamp_@bkamp_.tmi.twitch.tv PRIVMSG #botbkamp :!hey
[#botbkamp]> quit
double free or corruption (out)
Aborted (core dumped)
```

however the new command is still written to `commands.csv` so I mean, It works... 
if you don't want to or don't feel like doing this last change, just let me know.

File: cmdfile.c (hunk 1 above)
Comment (calebcscott) [reply cmdfile.c:86]: Typically errno is useful for functions that need to return information to caller but also want to indicate error status in some fashion, will change error return codes to macro defines to better show/standardize across file when error occurs

**Pull request closed without merging**